app.py

```diff
@@ -84,6 +99,19 @@ def delete_animal(pk: int) -> Union[Response, tuple[Response, int]]:
     return jsonify({"message": "Animal deleted successfully!"})
 
 
+@app.route('/search', methods=['GET'])  # Додали ендпоінт для пошуку за іменем
+def search_animal():
+    name = request.args.get('name', '')
+    animals = Animal.query.filter(Animal.name.like(f'%{name}%')).all()
+    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+
+
+def calculate_age(birth_date):
```

Comment:
I don't see where you use this function.
But I see that in sqlalchemy model you have property `age`.
I'd suggest to move this property to Pydantic model
https://docs.pydantic.dev/2.0/usage/computed_fields/
In this case it will be sent as part of context to frontend. So you can add it as new column in list table on HTML page. But you don't need it in add animal and update animal form (because it will be calculated each time and not stored in DB).

`age = today.year - self.birth_date.year` will not work properly. Because if date of birth will be 01.12.2022, it will show 1 year old, but really it's less than 1 year, so this code is better
`age = today.year - birth_date.year - ((today.month, today.day) < (birth_date.month, birth_date.day))`
Please move it to age property. And then you can delete function `def calculate_age(birth_date):`
as redundant.

Reply:
yes indeed I made a mistake, I will definitely change it